### PR TITLE
CI: Optimize CircleCI using a local docker registry instead docker save/load

### PR DESCRIPTION
**Pull request by oesteban**
Seems like replacing the docker save/load approach with a local registry run on the same machine allows us to make the main build job ~8min lighter and smoke tests builds ~6min lighter.

This is probably not a big deal time-wise, but I think this approach is more reliable than the docker save/load, and will escalate better if CircleCI opens up to having service jobs or some such.

This PR also revises the artifacts being collected - which were too many for successful runs (including the work directory and freesurfer folder).

The idea is to test how this works for sMRIPrep and then propagate to fMRIPrep and other projects if things look good.

### CODE REVIEW

**Comment by effigies** [comment]
I guess if it works, cool? I have no context for evaluating this apart from "Will the test pass".

**Comment by oesteban** [comment]
Yup, it seems to work, looking at the layer digests and manifests. My only concern is that it seems to keep slowly growing. - I'm looking right now at how to garbage collect unused layers.

**Comment by oesteban** [comment]
Nice - this https://stackoverflow.com/a/57989780 worked out locally. Testing now on Circle.

**Comment by oesteban** [comment]
Working on Circle too! - https://app.circleci.com/jobs/github/oesteban/smriprep/941/parallel-runs/0/steps/0-107

The step to archive and propagate the workspace went back below 5m as in the early runs where the registry was still clean.

If tests pass, we can go ahead :)

**Comment by oesteban** [comment]
Yass! Almost 1GB of difference after garbage collection:

- Last run (w/ cleaning up): [4.9GB / 2:54](https://app.circleci.com/jobs/github/oesteban/smriprep/946/parallel-runs/0/steps/0-101)
- Previous run (wo/ cleaning up): [5.8GB / 3:37](https://app.circleci.com/jobs/github/oesteban/smriprep/937/parallel-runs/0/steps/0-101)

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,30 +69,35 @@ jobs:
...
     working_directory: /tmp/src/smriprep
     steps:
       - attach_workspace:
           at: /tmp
       - save_cache:
-         key: build-v1-{{ .Branch }}-{{ epoch }}
+         key: build-v2-{{ .Branch }}-{{ epoch }}
          paths:
-            - /tmp/cache/docker.tar.gz
+            - /tmp/docker
```
Comment (effigies) [inline .circleci/config.yml]: In general I've found caches to be quicker, so there's no advantage to putting things in the workspace and then copying over to the cache in a separate job. Just purely use the cache.

Comment (effigies) [reply .circleci/config.yml]: See, e.g, https://github.com/poldracklab/fmriprep/pull/1608

**Comment by oesteban** [comment]
Okay, the full workflow is now under 30 min! - Merging.

**Pull request merged**